File: eKucniLjubimci.Services/Database/Artikal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eKucniLjubimci.Services.Database
{
    public class Artikal
    {
        public int ArtikalId { get; set; }
        public string Naziv { get; set; }
        public decimal Cijena { get; set; }
        public bool Dostupnost { get; set; }
        public string Opis { get; set; }        
        public string StateMachine { get; set; }

        public virtual List<NarudzbaArtikal> NarudzbeArtikli { get; set; }

        public int KategorijaId { get; set; }
        public virtual Kategorija Kategorija { get; set; }

        public virtual List<Slika> Slike { get; set; } = new List<Slika>();

    }
}

File: flutterUI/flutter_mobile/lib/models/artikal.dart
import 'kategorija.dart';
import 'package:flutter_mobile/models/slika.dart';
import 'package:json_annotation/json_annotation.dart';

part 'artikal.g.dart';

@JsonSerializable()
class Artikal {
  int? artikalId;
  String? naziv;
  double? cijena;
  bool? dostupnost;
  String? opis;
  String? stateMachine;
  int? kategorijaId;
  Kategorija? kategorija;
  List<Slika>? slike;

  Artikal({
    this.artikalId,
    this.naziv,
    this.cijena,
    this.dostupnost,
    this.opis,
    this.stateMachine,
    this.kategorijaId,
    this.kategorija,
    this.slike,
  });

  factory Artikal.fromJson(Map<String, dynamic> json) =>
      _$ArtikalFromJson(json);

  Map<String, dynamic> toJson() => _$ArtikalToJson(this);
}
